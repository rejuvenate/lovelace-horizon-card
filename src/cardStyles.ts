import { css } from 'lit'

export default css`
  :host {
    --hc-primary: var(--primary-text-color);
    --hc-secondary: var(--secondary-text-color);

    --hc-field-name-color: var(--hc-secondary);
    --hc-field-value-color: var(--hc-primary);

    --hc-day-color: #8ebeeb;
    --hc-night-color: #393b78;

    --hc-accent: #d7d7d7;
    --hc-lines: var(--hc-accent);

    --hc-sun-hue: 44;
    --hc-sun-saturation: 93%;
    --hc-sun-lightness: 67%;
    --hc-sun-hue-reduce: 0;
    --hc-sun-saturation-reduce: 0%;
    --hc-sun-lightness-reduce: 0%;
    --hc-sun-color: hsl(
      calc(var(--hc-sun-hue) - var(--hc-sun-hue-reduce)),
      calc(var(--hc-sun-saturation) - var(--hc-sun-saturation-reduce)),
      calc(var(--hc-sun-lightness) - var(--hc-sun-lightness-reduce))
    );

    --hc-moon-hue: 52;
    --hc-moon-saturation: 77%;
    --hc-moon-lightness: 57%;
    --hc-moon-saturation-reduce: 0%;
    --hc-moon-lightness-reduce: 0%;
    --hc-moon-color: hsl(
      var(--hc-moon-hue),
      calc(var(--hc-moon-saturation) - var(--hc-moon-saturation-reduce)),
      calc(var(--hc-moon-lightness) - var(--hc-moon-lightness-reduce))
    );
    --hc-moon-shadow-color: #eeeeee;
    --hc-moon-spot-color: rgba(170, 170, 170, 0.1);
  }

  :host(.horizon-card-dark) {
    --hc-accent: #464646;
    --hc-moon-saturation: 80%;
    --hc-moon-lightness: 74%;
    --hc-moon-shadow-color: #272727;
  }

  .horizon-card {
    padding: 0.5em;
    font-family: var(--primary-font-family);
  }

  .horizon-card-field-row {
    display: flex;
    justify-content: space-around;
    margin-top: 1em;
    margin-bottom: -0.3em;
  }

  .horizon-card-text-container {
    display: flex;
    flex-direction: column;
    align-items: center;
  }

  .horizon-card-field-name {
    color: var(--hc-field-name-color);
  }

  .horizon-card-field-value {
    color: var(--hc-field-value-color);
    font-size: 1.2em;
    line-height: 1.1em;
    text-align: center;
  }

  .horizon-card-field-value-moon-phase {
    font-size: inherit;
  }

  .horizon-card-field-moon-phase {
    --mdc-icon-size: 2em;
    color: var(--primary-color);
  }

  .horizon-card-field-value-secondary {
    font-size: 0.7em;
  }

  .horizon-card-sun-value:before {
    content: "☉";
    padding-right: 0.5em;
  }

  .horizon-card-moon-value:before {
    content: "☽";
    padding-right: 0.5em;
  }

  .horizon-card-header {
    display: flex;
    justify-content: space-around;
    margin-top: 1em;
    margin-bottom: -0.3em;
  }

  .horizon-card-header .horizon-card-text-container {
    font-size: 1.2em;
  }

  .horizon-card-footer {
    margin-bottom: 1em;
  }

  .horizon-card-title {
    margin: 1em 1em 1em 1em;
    font-size: 1.5em;
    color: var(--hc-primary);
  }

  .horizon-card-graph {
    margin: 1em 0.5em 1em 0.5em;
  }

  .horizon-card-graph .dawn {
    fill: var(--hc-night-color);
    stroke: var(--hc-night-color);
  }

  .horizon-card-graph .day {
    fill: var(--hc-day-color);
    stroke: var(--hc-day-color);
  }
`
