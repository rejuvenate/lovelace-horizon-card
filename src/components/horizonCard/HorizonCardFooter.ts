import { html, nothing, TemplateResult } from 'lit'

import {
  EHorizonCardI18NKeys,
  IHorizonCardConfig,
  THorizonCardData,
  THorizonCardFields,
  TMoonTimes,
  TSunTimes
} from '../../types'
import { HelperFunctions } from '../../utils/HelperFunctions'
import { I18N } from '../../utils/I18N'

export class HorizonCardFooter {
  private readonly data: THorizonCardData
  private readonly i18n: I18N
  private readonly sunTimes: TSunTimes
  private readonly moonTimes: TMoonTimes
  private readonly fields: THorizonCardFields
  private readonly azimuths
  private readonly azimuthExtraClasses: string[]
  private readonly elevations
  private readonly elevationExtraClasses: string[]

  constructor (config: IHorizonCardConfig, data: THorizonCardData, i18n: I18N) {
    this.data = data

    this.i18n = i18n
    this.sunTimes = data.sunData.times
    this.moonTimes = data.moonData.times
    // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
    this.fields = config.fields!

    this.azimuths = []
    if (this.fields.sun_azimuth) {
      this.azimuths.push(this.data.sunData.azimuth)
    }
    if (this.fields.moon_azimuth) {
      this.azimuths.push(this.data.moonData.azimuth)
    }
    if (this.fields.sun_azimuth && this.fields.moon_azimuth) {
      this.azimuthExtraClasses = ['horizon-card-sun-value', 'horizon-card-moon-value']
    } else {
      this.azimuthExtraClasses = []
    }

    this.elevations = []
    if (this.fields.sun_elevation) {
      this.elevations.push(this.data.sunData.elevation)
    }
    if (this.fields.moon_elevation) {
      this.elevations.push(this.data.moonData.elevation)
    }
    if (this.fields.sun_elevation && this.fields.moon_elevation) {
      this.elevationExtraClasses = ['horizon-card-sun-value', 'horizon-card-moon-value']
    } else {
      this.elevationExtraClasses = []
    }
  }

  public render (): TemplateResult {
    return html`
      <div class="horizon-card-footer">
        ${
          this.renderRow(
            this.fields.dawn
              ? HelperFunctions.renderFieldElement(this.i18n, EHorizonCardI18NKeys.Dawn, this.sunTimes.dawn)
              : nothing,
            this.fields.noon
              ? HelperFunctions.renderFieldElement(this.i18n, EHorizonCardI18NKeys.Noon, this.sunTimes.noon)
              : nothing,
            this.fields.dusk
              ? HelperFunctions.renderFieldElement(this.i18n, EHorizonCardI18NKeys.Dusk, this.sunTimes.dusk)
              : nothing
          )
        }
        ${
          this.renderRow(
            this.fields.sun_azimuth || this.fields.moon_azimuth
              ? HelperFunctions.renderFieldElements(this.i18n, EHorizonCardI18NKeys.Azimuth, this.azimuths,
                this.azimuthExtraClasses)
              : nothing,
            this.fields.sun_elevation || this.fields.moon_elevation
              ? HelperFunctions.renderFieldElements(this.i18n, EHorizonCardI18NKeys.Elevation, this.elevations,
                this.elevationExtraClasses)
              : nothing
          )
        }
        ${
          this.renderRow(
            this.fields.moonrise
              ? HelperFunctions.renderFieldElement(this.i18n, EHorizonCardI18NKeys.Moonrise, this.moonTimes.moonrise)
              : nothing,
            this.fields.moon_phase
              ? HelperFunctions.renderMoonElement(this.i18n, this.data.moonData.phase, this.data.moonData.phaseRotation)
              : nothing,
            this.fields.moonset
              ? HelperFunctions.renderFieldElement(this.i18n, EHorizonCardI18NKeys.Moonset, this.moonTimes.moonset)
              : nothing
          )
        }
      </div>
    `
  }

  private renderRow (...args: (typeof nothing | TemplateResult)[]) {
    const nonEmpty = args.filter((tr) => tr !== nothing)
    return nonEmpty.length > 0
      ? html`
        <div class="horizon-card-field-row">
          ${nonEmpty}
        </div>`
      : nothing
  }
}
