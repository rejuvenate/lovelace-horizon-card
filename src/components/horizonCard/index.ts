export * from './HorizonCard'
export * from './HorizonCardContent'
export * from './HorizonCardFooter'
export * from './HorizonCardGraph'
export * from './HorizonCardHeader'
