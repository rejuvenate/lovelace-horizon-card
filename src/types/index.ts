import { LovelaceCardConfig, NumberFormat, TimeFormat } from 'custom-card-helpers'

export type THorizonCardFields = {
  sunrise?: boolean
  sunset?: boolean

  dawn?: boolean
  noon?: boolean
  dusk?: boolean

  azimuth?: boolean
  sun_azimuth?: boolean
  moon_azimuth?: boolean
  elevation?: boolean
  sun_elevation?: boolean
  moon_elevation?: boolean

  moonrise?: boolean,
  moonset?: boolean,
  moon_phase?: boolean
}

export interface IHorizonCardConfig extends LovelaceCardConfig {
  dark_mode?: boolean
  language?: string

  title?: string,
  moon?: boolean,
  time_format?: TimeFormat
  number_format?: NumberFormat

  latitude?: number
  longitude?: number
  elevation?: number
  southern_flip?: boolean
  moon_phase_rotation?: number
  now?: Date
  time_zone?: string
  refresh_period?: number
  debug_level?: number

  fields?: THorizonCardFields
}

export type TSunTimes = {
  now: Date
  dawn?: Date
  dusk?: Date
  midnight: Date
  noon: Date
  sunrise?: Date
  sunset?: Date
}

export type TSunData = {
  readonly azimuth: number
  readonly elevation: number
  readonly times: TSunTimes
  readonly hueReduce: number
  readonly saturationReduce: number
  readonly lightnessReduce: number
}

export type TSunPosition = {
  readonly x: number
  readonly y: number
  readonly horizonY: number
  readonly sunriseX: number
  readonly sunsetX: number
  readonly scaleY: number
  readonly offsetY: number
}

export type TMoonTimes = {
  readonly now: Date,
  readonly moonrise: Date,
  readonly moonset: Date
}

export type TMoonPhase = {
  state: string
  icon: string
}

export type SunCalcMoonPhase =
  'newMoon'
  | 'waxingCrescentMoon'
  | 'firstQuarterMoon'
  | 'waxingGibbousMoon'
  | 'fullMoon'
  | 'waningGibbousMoon'
  | 'thirdQuarterMoon'
  | 'waningCrescentMoon'

export type TMoonData = {
  readonly azimuth: number
  readonly elevation: number
  readonly fraction: number
  readonly phase: TMoonPhase
  readonly phaseRotation: number,
  readonly zenithAngle: number
  readonly parallacticAngle: number
  readonly times: TMoonTimes,
  readonly saturationReduce: number,
  readonly lightnessReduce: number
}

export type TMoonPosition = {
  readonly x: number
  readonly y: number
}

export enum EHorizonCardErrors {
}

export type THorizonCardData = {
  readonly partial: boolean
  readonly latitude: number
  readonly longitude: number
  readonly sunPosition: TSunPosition
  readonly sunData: TSunData
  readonly moonPosition: TMoonPosition
  readonly moonData: TMoonData
}

export enum EHorizonCardI18NKeys {
  Azimuth = 'azimuth',
  Dawn = 'dawn',
  Dusk = 'dusk',
  Elevation = 'elevation',
  Noon = 'noon',
  Sunrise = 'sunrise',
  Sunset = 'sunset',
  Moonrise = 'moonrise',
  Moonset = 'moonset'
}

export type THorizonCardI18NErrorKeys = {
  [key in EHorizonCardErrors]: string
}

export type THorizonCardI18NKeys = { [key in EHorizonCardI18NKeys ]?: string } | { errors: THorizonCardI18NErrorKeys }
