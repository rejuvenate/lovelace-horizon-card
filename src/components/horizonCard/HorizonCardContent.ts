import { html, TemplateResult } from 'lit'

import { IHorizonCardConfig,THorizonCardData } from '../../types'
import { I18N } from '../../utils/I18N'
import { HorizonCardFooter } from './HorizonCardFooter'
import { HorizonCardGraph } from './HorizonCardGraph'
import { HorizonCardHeader } from './HorizonCardHeader'

export class HorizonCardContent {
  private config: IHorizonCardConfig
  private data: THorizonCardData
  private i18n: I18N

  constructor (config: IHorizonCardConfig, data: THorizonCardData, i18n: I18N) {
    this.config = config
    this.data = data
    this.i18n = i18n
  }

  render (): TemplateResult {
    return html`
      <ha-card>
        <div class="horizon-card">
          ${ this.renderHeader() }
          ${ this.renderGraph() }
          ${ this.renderFooter() }
        </div>
      </ha-card>
    `
  }

  private renderHeader (): TemplateResult {
    return new HorizonCardHeader(this.config, this.data, this.i18n).render()
  }

  private renderGraph (): TemplateResult {
    return new HorizonCardGraph(this.config, this.data).render()
  }

  private renderFooter (): TemplateResult {
    return new HorizonCardFooter(this.config, this.data, this.i18n).render()
  }
}
