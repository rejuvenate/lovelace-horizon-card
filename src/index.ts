import './components/horizonCard'
import './components/horizonCardEditor'

