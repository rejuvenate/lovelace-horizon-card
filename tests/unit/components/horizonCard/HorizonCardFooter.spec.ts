import { HorizonCardFooter } from '../../../../src/components/horizonCard/HorizonCardFooter'
import { IHorizonCardConfig, THorizonCardData } from '../../../../src/types'
import { CustomSnapshotSerializer, TemplateResultTestHelper } from '../../../helpers/TestHelpers'

jest.mock('../../../../src/utils/HelperFunctions', () => require('../../../mocks/HelperFunctions'))

expect.addSnapshotSerializer(new CustomSnapshotSerializer())

describe('HorizonCardFooter', () => {
  const dateFields = ['dawn', 'noon', 'dusk']
  const numberFields = ['azimuth', 'elevation']
  const getConfigForValue = (field: string) => ({ fields: { [field]: true } })
  const getDataForValue = (field: string) => {
    if (dateFields.includes(field)) {
      return {
        times: {
          [field]: new Date(0)
        }
      }
    }

    if (numberFields.includes(field)) {
      return {
        [field]: 0
      }
    }

    return
  }

  describe('render', () => {
    for (const field of [...dateFields, ...numberFields]) {
      it(`should render the ${field} field when it is present in the config and data for it is provided`, async () => {
        const config = getConfigForValue(field) as IHorizonCardConfig
        const data = getDataForValue(field) as THorizonCardData

        const horizonCardFooter = new HorizonCardFooter(config, data)
        const element = window.document.createElement('test-element') as TemplateResultTestHelper<typeof horizonCardFooter.render>
        element.templateResultFunction = () => horizonCardFooter.render()
        window.document.body.appendChild(element)
        await element.updateComplete

        expect(element.shadowRoot!.innerHTML).toMatchSnapshot()
      })

      it(`should not render the ${field} field when it is present in the config but no data for it is provided`, async () => {
        const config = getConfigForValue(field) as IHorizonCardConfig
        const data = {} as THorizonCardData

        const horizonCardFooter = new HorizonCardFooter(config, data)
        const element = window.document.createElement('test-element') as TemplateResultTestHelper<typeof horizonCardFooter.render>
        element.templateResultFunction = () => horizonCardFooter.render()
        window.document.body.appendChild(element)
        await element.updateComplete

        expect(element.shadowRoot!.innerHTML).toMatchSnapshot()
      })

      it(`should not render the ${field} field when it is not present in the config but data for it is provided`, async () => {
        const config = {} as IHorizonCardConfig
        const data = getDataForValue(field) as THorizonCardData

        const horizonCardFooter = new HorizonCardFooter(config, data)
        const element = window.document.createElement('test-element') as TemplateResultTestHelper<typeof horizonCardFooter.render>
        element.templateResultFunction = () => horizonCardFooter.render()
        window.document.body.appendChild(element)
        await element.updateComplete

        expect(element.shadowRoot!.innerHTML).toMatchSnapshot()
      })
    }
  })
})
