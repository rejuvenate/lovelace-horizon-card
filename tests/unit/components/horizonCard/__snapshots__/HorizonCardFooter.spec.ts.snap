// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HorizonCardFooter render should not render the dawn field when it is not present in the config but data for it is provided 1`] = `
<div class="horizon-card-footer">
</div>
`;

exports[`HorizonCardFooter render should not render the dusk field when it is not present in the config but data for it is provided 1`] = `
<div class="horizon-card-footer">
</div>
`;

exports[`HorizonCardFooter render should not render the moon_azimuth field when it is not present in the config but data for it is provided 1`] = `
<div class="horizon-card-footer">
</div>
`;

exports[`HorizonCardFooter render should not render the moon_elevation field when it is not present in the config but data for it is provided 1`] = `
<div class="horizon-card-footer">
</div>
`;

exports[`HorizonCardFooter render should not render the moon_phase field when it is not present in the config but data for it is provided 1`] = `
<div class="horizon-card-footer">
</div>
`;

exports[`HorizonCardFooter render should not render the moonrise field when it is not present in the config but data for it is provided 1`] = `
<div class="horizon-card-footer">
</div>
`;

exports[`HorizonCardFooter render should not render the moonset field when it is not present in the config but data for it is provided 1`] = `
<div class="horizon-card-footer">
</div>
`;

exports[`HorizonCardFooter render should not render the noon field when it is not present in the config but data for it is provided 1`] = `
<div class="horizon-card-footer">
</div>
`;

exports[`HorizonCardFooter render should not render the sun_azimuth field when it is not present in the config but data for it is provided 1`] = `
<div class="horizon-card-footer">
</div>
`;

exports[`HorizonCardFooter render should not render the sun_azimuth,moon_azimuth field when it is not present in the config but data for it is provided 1`] = `
<div class="horizon-card-footer">
</div>
`;

exports[`HorizonCardFooter render should not render the sun_elevation field when it is not present in the config but data for it is provided 1`] = `
<div class="horizon-card-footer">
</div>
`;

exports[`HorizonCardFooter render should not render the sun_elevation,moon_elevation field when it is not present in the config but data for it is provided 1`] = `
<div class="horizon-card-footer">
</div>
`;

exports[`HorizonCardFooter render should render the dawn field as '-' when it is present in the config but no data for it is provided 1`] = `
<div class="horizon-card-footer">
  <div class="horizon-card-field-row">
    <div class="horizon-card-text-container">
      <div class="horizon-card-field-name">
        dawn
      </div>
      <div class="horizon-card-field-value ">
        -
      </div>
    </div>
  </div>
</div>
`;

exports[`HorizonCardFooter render should render the dawn field when it is present in the config and data for it is provided 1`] = `
<div class="horizon-card-footer">
  <div class="horizon-card-field-row">
    <div class="horizon-card-text-container">
      <div class="horizon-card-field-name">
        dawn
      </div>
      <div class="horizon-card-field-value ">
        0
      </div>
    </div>
  </div>
</div>
`;

exports[`HorizonCardFooter render should render the dusk field as '-' when it is present in the config but no data for it is provided 1`] = `
<div class="horizon-card-footer">
  <div class="horizon-card-field-row">
    <div class="horizon-card-text-container">
      <div class="horizon-card-field-name">
        dusk
      </div>
      <div class="horizon-card-field-value ">
        -
      </div>
    </div>
  </div>
</div>
`;

exports[`HorizonCardFooter render should render the dusk field when it is present in the config and data for it is provided 1`] = `
<div class="horizon-card-footer">
  <div class="horizon-card-field-row">
    <div class="horizon-card-text-container">
      <div class="horizon-card-field-name">
        dusk
      </div>
      <div class="horizon-card-field-value ">
        0
      </div>
    </div>
  </div>
</div>
`;

exports[`HorizonCardFooter render should render the moon_azimuth field as '-' when it is present in the config but no data for it is provided 1`] = `
<div class="horizon-card-footer">
  <div class="horizon-card-field-row">
    <div class="horizon-card-text-container">
      <div class="horizon-card-field-name">
        azimuth
      </div>
      <div class="horizon-card-field-value ">
        -
      </div>
    </div>
  </div>
</div>
`;

exports[`HorizonCardFooter render should render the moon_azimuth field when it is present in the config and data for it is provided 1`] = `
<div class="horizon-card-footer">
  <div class="horizon-card-field-row">
    <div class="horizon-card-text-container">
      <div class="horizon-card-field-name">
        azimuth
      </div>
      <div class="horizon-card-field-value ">
        0°
      </div>
    </div>
  </div>
</div>
`;

exports[`HorizonCardFooter render should render the moon_elevation field as '-' when it is present in the config but no data for it is provided 1`] = `
<div class="horizon-card-footer">
  <div class="horizon-card-field-row">
    <div class="horizon-card-text-container">
      <div class="horizon-card-field-name">
        elevation
      </div>
      <div class="horizon-card-field-value ">
        -
      </div>
    </div>
  </div>
</div>
`;

exports[`HorizonCardFooter render should render the moon_elevation field when it is present in the config and data for it is provided 1`] = `
<div class="horizon-card-footer">
  <div class="horizon-card-field-row">
    <div class="horizon-card-text-container">
      <div class="horizon-card-field-name">
        elevation
      </div>
      <div class="horizon-card-field-value ">
        0°
      </div>
    </div>
  </div>
</div>
`;

exports[`HorizonCardFooter render should render the moon_phase field as '-' when it is present in the config but no data for it is provided 1`] = `
<div class="horizon-card-footer">
</div>
`;

exports[`HorizonCardFooter render should render the moon_phase field when it is present in the config and data for it is provided 1`] = `
<div class="horizon-card-footer">
  <div class="horizon-card-field-row">
    <div class="horizon-card-text-container">
      <div class="horizon-card-field-moon-phase"
           style="transform: rotate(0deg)"
      >
        <ha-icon icon="mdi:">
        </ha-icon>
      </div>
      <div class="horizon-card-field-value horizon-card-field-value-moon-phase">
        component.sensor.state.moon__phase.undefined
      </div>
    </div>
  </div>
</div>
`;

exports[`HorizonCardFooter render should render the moonrise field as '-' when it is present in the config but no data for it is provided 1`] = `
<div class="horizon-card-footer">
  <div class="horizon-card-field-row">
    <div class="horizon-card-text-container">
      <div class="horizon-card-field-name">
        moonrise
      </div>
      <div class="horizon-card-field-value ">
        -
      </div>
    </div>
  </div>
</div>
`;

exports[`HorizonCardFooter render should render the moonrise field when it is present in the config and data for it is provided 1`] = `
<div class="horizon-card-footer">
  <div class="horizon-card-field-row">
    <div class="horizon-card-text-container">
      <div class="horizon-card-field-name">
        moonrise
      </div>
      <div class="horizon-card-field-value ">
        0
      </div>
    </div>
  </div>
</div>
`;

exports[`HorizonCardFooter render should render the moonset field as '-' when it is present in the config but no data for it is provided 1`] = `
<div class="horizon-card-footer">
  <div class="horizon-card-field-row">
    <div class="horizon-card-text-container">
      <div class="horizon-card-field-name">
        moonset
      </div>
      <div class="horizon-card-field-value ">
        -
      </div>
    </div>
  </div>
</div>
`;

exports[`HorizonCardFooter render should render the moonset field when it is present in the config and data for it is provided 1`] = `
<div class="horizon-card-footer">
  <div class="horizon-card-field-row">
    <div class="horizon-card-text-container">
      <div class="horizon-card-field-name">
        moonset
      </div>
      <div class="horizon-card-field-value ">
        0
      </div>
    </div>
  </div>
</div>
`;

exports[`HorizonCardFooter render should render the noon field as '-' when it is present in the config but no data for it is provided 1`] = `
<div class="horizon-card-footer">
  <div class="horizon-card-field-row">
    <div class="horizon-card-text-container">
      <div class="horizon-card-field-name">
        noon
      </div>
      <div class="horizon-card-field-value ">
        -
      </div>
    </div>
  </div>
</div>
`;

exports[`HorizonCardFooter render should render the noon field when it is present in the config and data for it is provided 1`] = `
<div class="horizon-card-footer">
  <div class="horizon-card-field-row">
    <div class="horizon-card-text-container">
      <div class="horizon-card-field-name">
        noon
      </div>
      <div class="horizon-card-field-value ">
        0
      </div>
    </div>
  </div>
</div>
`;

exports[`HorizonCardFooter render should render the sun_azimuth field as '-' when it is present in the config but no data for it is provided 1`] = `
<div class="horizon-card-footer">
  <div class="horizon-card-field-row">
    <div class="horizon-card-text-container">
      <div class="horizon-card-field-name">
        azimuth
      </div>
      <div class="horizon-card-field-value ">
        -
      </div>
    </div>
  </div>
</div>
`;

exports[`HorizonCardFooter render should render the sun_azimuth field when it is present in the config and data for it is provided 1`] = `
<div class="horizon-card-footer">
  <div class="horizon-card-field-row">
    <div class="horizon-card-text-container">
      <div class="horizon-card-field-name">
        azimuth
      </div>
      <div class="horizon-card-field-value ">
        0°
      </div>
    </div>
  </div>
</div>
`;

exports[`HorizonCardFooter render should render the sun_azimuth,moon_azimuth field as '-' when it is present in the config but no data for it is provided 1`] = `
<div class="horizon-card-footer">
  <div class="horizon-card-field-row">
    <div class="horizon-card-text-container">
      <div class="horizon-card-field-name">
        azimuth
      </div>
      <div class="horizon-card-field-value horizon-card-sun-value">
        -
      </div>
      <div class="horizon-card-field-value horizon-card-moon-value">
        -
      </div>
    </div>
  </div>
</div>
`;

exports[`HorizonCardFooter render should render the sun_azimuth,moon_azimuth field when it is present in the config and data for it is provided 1`] = `
<div class="horizon-card-footer">
  <div class="horizon-card-field-row">
    <div class="horizon-card-text-container">
      <div class="horizon-card-field-name">
        azimuth
      </div>
      <div class="horizon-card-field-value horizon-card-sun-value">
        0°
      </div>
      <div class="horizon-card-field-value horizon-card-moon-value">
        0°
      </div>
    </div>
  </div>
</div>
`;

exports[`HorizonCardFooter render should render the sun_elevation field as '-' when it is present in the config but no data for it is provided 1`] = `
<div class="horizon-card-footer">
  <div class="horizon-card-field-row">
    <div class="horizon-card-text-container">
      <div class="horizon-card-field-name">
        elevation
      </div>
      <div class="horizon-card-field-value ">
        -
      </div>
    </div>
  </div>
</div>
`;

exports[`HorizonCardFooter render should render the sun_elevation field when it is present in the config and data for it is provided 1`] = `
<div class="horizon-card-footer">
  <div class="horizon-card-field-row">
    <div class="horizon-card-text-container">
      <div class="horizon-card-field-name">
        elevation
      </div>
      <div class="horizon-card-field-value ">
        0°
      </div>
    </div>
  </div>
</div>
`;

exports[`HorizonCardFooter render should render the sun_elevation,moon_elevation field as '-' when it is present in the config but no data for it is provided 1`] = `
<div class="horizon-card-footer">
  <div class="horizon-card-field-row">
    <div class="horizon-card-text-container">
      <div class="horizon-card-field-name">
        elevation
      </div>
      <div class="horizon-card-field-value horizon-card-sun-value">
        -
      </div>
      <div class="horizon-card-field-value horizon-card-moon-value">
        -
      </div>
    </div>
  </div>
</div>
`;

exports[`HorizonCardFooter render should render the sun_elevation,moon_elevation field when it is present in the config and data for it is provided 1`] = `
<div class="horizon-card-footer">
  <div class="horizon-card-field-row">
    <div class="horizon-card-text-container">
      <div class="horizon-card-field-name">
        elevation
      </div>
      <div class="horizon-card-field-value horizon-card-sun-value">
        0°
      </div>
      <div class="horizon-card-field-value horizon-card-moon-value">
        0°
      </div>
    </div>
  </div>
</div>
`;
